nc/include/component/Transform.h

```diff
@@ -5,51 +5,64 @@
 #include "directx/math/DirectXMath.h"
 
+#include <span>
+#include <vector>
+
 namespace nc
 {
-    enum class Space : uint8_t
-    {
-        World,
-        Local
-    };
-
     class Transform final : public ComponentBase
     {
         public:
-            Transform(EntityHandle handle, const Vector3& pos, const Quaternion& rot, const Vector3& scale);
+            Transform(EntityHandle handle, const Vector3& pos, const Quaternion& rot, const Vector3& scale, Transform* parent);
             ~Transform() = default;
             Transform(const Transform&) = delete;
             Transform(Transform&&) = delete;
             Transform& operator=(const Transform&) = delete;
             Transform& operator=(Transform&&) = delete;
 
-            Vector3 GetPosition() const;
-            Quaternion GetRotation() const;
-            Vector3 GetScale() const;
-            DirectX::FXMMATRIX GetTransformationMatrix() const;
-            DirectX::XMMATRIX GetTransformationMatrixEx(Vector3 additionalScale) const;
-            DirectX::XMMATRIX GetViewMatrix() const;
-            Vector3 ToLocalSpace(const Vector3& vec) const;
-            Vector3 Up() const;
-            Vector3 Forward() const;
-            Vector3 Right() const;
+            Vector3 GetPosition() const;                        // Get world space position
+            Vector3 GetLocalPosition() const;                   // Get position relative to parent
+            Quaternion GetRotation() const;                     // Get world space rotation
+            Quaternion GetLocalRotation() const;                // Get rotation relative to parent
+            Vector3 GetScale() const;                           // Get world space scale
+            Vector3 GetLocalScale() const;                      // Get scale relative to parent
+            DirectX::FXMMATRIX GetTransformationMatrix() const; // Get world space transformation matrix
+            Vector3 ToLocalSpace(const Vector3& vec) const;     // Transform a vector by this transforms's world space matrix
+            Vector3 Up() const;                                 // Get the up axis of the transform
+            Vector3 Forward() const;                            // Get the forward axis of the transform
+            Vector3 Right() const;                              // Get the right axis of the transform
 
+            // All modification methods operate in local space
             void Set(const Vector3& pos, const Quaternion& quat, const Vector3& scale);
             void Set(const Vector3& pos, const Vector3& angles, const Vector3& scale);
             void SetPosition(const Vector3& pos);
             void SetRotation(const Quaternion& quat);
             void SetRotation(const Vector3& angles);
             void SetScale(const Vector3& scale);
-
-            void Translate(Vector3 vec, Space space = Space::Local);
+            void Translate(Vector3 vec);
+            void TranslateLocalSpace(Vector3 vec);
```

Comment:
What's going on with this method and the above comment? Does Translate(Vector3 vec) operate in world space then? If so, we should name all other local space setters XLocalSpace or XLocal